pkg/validate/security_context.go

```diff
@@ -450,8 +502,123 @@ var _ = framework.KubeDescribe("Security Context", func() {
 		})
-	})
 
+		It("runtime should support an seccomp profile that blocks setting hostname with no sysCaps", func() {
+			privileged := false
+			expectContainerCreateToPass := true
+			By("create pod")
+			podID, podConfig = framework.CreatePodSandboxForContainer(rc)
+			By("create container with seccompBlockHostNameProfile and test")
+			containerID := createSeccompContainer(rc, ic, podID, podConfig,
+				"container-with-block-hostname-seccomp-profile-test-",
+				localhost+blockHostNameProfilePath, nil, privileged, expectContainerCreateToPass)
+			startContainer(rc, containerID)
+			Eventually(func() runtimeapi.ContainerState {
+				return getContainerStatus(rc, containerID).State
+			}, time.Minute, time.Second*4).Should(Equal(runtimeapi.ContainerState_CONTAINER_RUNNING))
+			checkSetHostname(rc, containerID, false)
+		})
+
+		It("runtime should not support a custom seccomp profile without using localhost/ as a prefix", func() {
+			privileged := false
+			expectContainerCreateToPass := false
+			By("create pod")
+			podID, podConfig = framework.CreatePodSandboxForContainer(rc)
+			By("create container with seccompBlockHostNameProfile and test")
+			_ = createSeccompContainer(rc, ic, podID, podConfig,
+				"container-with-block-hostname-seccomp-profile-test-",
+				blockHostNameProfilePath, nil, privileged, expectContainerCreateToPass)
+		})
+
+		It("runtime should ignore a seccomp profile that blocks setting hostname when privileged", func() {
+			privileged := true
+			expectContainerCreateToPass := true
+			By("create privileged pod")
+			podID, podConfig = createPrivilegedPodSandbox(rc, true)
+			By("create privileged container with seccompBlockHostNameProfile and test")
+			containerID := createSeccompContainer(rc, ic, podID, podConfig,
+				"container-with-block-hostname-seccomp-profile-test-",
+				localhost+blockHostNameProfilePath, nil, privileged, expectContainerCreateToPass)
+			startContainer(rc, containerID)
+			Eventually(func() runtimeapi.ContainerState {
+				return getContainerStatus(rc, containerID).State
+			}, time.Minute, time.Second*4).Should(Equal(runtimeapi.ContainerState_CONTAINER_RUNNING))
+			checkSetHostname(rc, containerID, true)
+		})
+
+		It("runtime should block setting hostname with no seccomp profile specified", func() {
+			By("create pod")
+			podID, podConfig = framework.CreatePodSandboxForContainer(rc)
+			By("create container without seccomp profile and test")
+			containerID := framework.CreateDefaultContainer(rc, ic, podID, podConfig,
+				"container-with-no-seccomp-profile-test-")
+			startContainer(rc, containerID)
+			Eventually(func() runtimeapi.ContainerState {
+				return getContainerStatus(rc, containerID).State
+			}, time.Minute, time.Second*4).Should(Equal(runtimeapi.ContainerState_CONTAINER_RUNNING))
+			checkSetHostname(rc, containerID, false)
+		})
+
+		It("runtime should support setting hostname with docker/default seccomp profile and sysCaps", func() {
```

Comment:
Move this to a separate `docker/default` context.

Reply:
Yes, we both had that idea at the same time :-)